### PR TITLE
[Feature] Add tensorrt evaluation

### PR DESCRIPTION
**Pull request by AllentDan**
Mainly changes the following files:
.gitignore: add onnx and tensorrt file and local history
docs/tools_scripts.md: add tensorrt evaluation results
tools/deploy_test.py: add tensorrt evaluation parts

### CODE REVIEW

File: docs/tools_scripts.md [hunk 1]
```diff
@@ -312,3 +151,202 @@ The table below lists the models that are guaranteed to be exportable to TensorR
 - *All models above are tested with Pytorch==1.8.1,  onnxruntime==1.7.0 and tensorrt==7.2.3.4*
 - If you meet any problem with the listed models above, please create an issue and it would be taken care of soon. For models not included in the list, please try to solve them by yourself.
 - Because this feature is experimental and may change fast, please always try with the latest `mmcv` and `mmedit`.
+
+### Evaluate ONNX and TensorRT Models (experimental)
+
+We provide methods to evaluate TensorRT and ONNX models in `tools/deploy_test.py`. To evaluate ONNX models, onnxruntime should be installed first.
+
+#### Prerequisite
```
Comment (RunningLeon) [inline docs/tools_scripts.md:159]: We may need to install TensorRT as well for the evaluation of TensorRT models.

File: docs/tools_scripts.md [hunk 2]
```diff
@@ -312,3 +151,202 @@ The table below lists the models that are guaranteed to be exportable to TensorR
...
+- `config`: The path of a model config file.
+- `model`: The path of a TensorRT or an ONNX model file.
+- `backend`: The backend for testing, choose tensorrt or onnxruntime.
+- `--out`: The path of output result file in pickle format.
+- `--save-path`: The path to store images and if not given, it will not save image.
+- `--cfg-options`: Override some settings in the used config file, the key-value pair in `xxx=yyy` format will be merged into config file.
+
+#### Results and Models
+
+<table class="tg">
```
Comment (RunningLeon) [inline docs/tools_scripts.md:185]: Please keep the format of html table the same as before. For instance, the `class` attribute.
You can check generated HTML pages by following instructions:
- Generate HTML pages.
```bash
pip install recommonmark sphinx_markdown_tables sphinx_rtd_theme
cd docs
make html
```
- Then view the related HTML files with the browser.

Comment (AllentDan) [reply docs/tools_scripts.md:185]: Well, I used [Table Generator](https://www.tablesgenerator.com/html_tables). But your advise will be taken.

File: docs/tools_scripts.md [hunk 3]
```diff
@@ -99,24 +99,73 @@ The table below lists the models that are guaranteed to be exportable to ONNX an
 - If you meet any problem with the listed models above, please create an issue and it would be taken care of soon. For models not included in the list, please try to solve them by yourself.
 - Because this feature is experimental and may change fast, please always try with the latest `mmcv` and `mmedit`.
 
-### Evaluate ONNX Models with ONNXRuntime (experimental)
+### Convert ONNX to TensorRT (experimental)
+We also provide a script to convert [ONNX](https://github.com/onnx/onnx) model to [TensorRT](https://github.com/NVIDIA/TensorRT) format. Besides, we support comparing the output results between ONNX and TensorRT model.
```
Comment (innerlee) [inline docs/tools_scripts.md:103]: Add an empty line between section heading and content

File: tools/deploy_test.py [hunk 4]
```diff
@@ -1,18 +1,97 @@
...
+class TensorRTRestorerGenerator(nn.Module):
+    """Inner class for tensorrt restorer model inference
+
+    Args:
+        trt_file (str): The path to the tensorrt file.
+        device_id (int): Which device to place the model.
+    """
+
+    def __init__(self, trt_file: str, device_id: int):
+        super(TensorRTRestorerGenerator, self).__init__()
```
Comment (innerlee) [inline tools/deploy_test.py:26]: `super().__init__()`

File: tools/deploy_test.py [hunk 5]
```diff
@@ -1,18 +1,97 @@
...
+    """A warper class for tensorrt restorer
+
+    Args:
+        base_model (Any): The base model build from config.
+        trt_file (str): The path to the tensorrt file.
+        device_id (int): Which device to place the model.
+    """
+
+    def __init__(self, base_model: Any, trt_file: str, device_id: int):
+        super(TensorRTRestorer, self).__init__()
```
Comment (innerlee) [inline tools/deploy_test.py:55]: super

**Review by innerlee — APPROVED** [review]

**Comment by innerlee** [comment]
Some minor style change

**Pull request merged**